src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -196,13 +199,12 @@ def __enter__(self):
         self._inputs_queue = Queue()
         # Starting a new process in non-fork mode requires to allocate memory. Determine the maximum number of processes
         # based on available memory to avoid memory bursting.
-        if not self._use_fork:
-            available_max_worker_count = get_available_max_worker_count()
-            self._n_process = min(self._worker_count, self._nlines, available_max_worker_count)
-            bulk_logger.info(f"Not using fork, process count: {self._n_process}")
+        if self._use_default_worker_count:
+            self._n_process = self._determine_worker_count()
         else:
-            self._n_process = min(self._worker_count, self._nlines)
-            bulk_logger.info(f"Using fork, process count: {self._n_process}")
+            self._n_process = self._worker_count
+            self._log_process_count_info()
```

Comment:
We can also move this `if .. else .. ` block into `self._determine_worker_count()`, since we want it to determine the worker count by checking various conditions and only give us a result here.

Reply:
fixed